extensions/amp-video/0.1/amp-video.js

```diff
@@ -259,11 +260,20 @@ export class AmpVideo extends AMP.BaseElement {
     // Cached so mediapool operations (eg: swapping sources) don't interfere with this bool.
     this.hasBitrateSources_ =
       !!this.element.querySelector('source[data-bitrate]') ||
-      this.hasAnyCachedSources_();
+      this.hasAnyCachedSources_() ||
+      this.element.hasAttribute('cache');
 
     installVideoManagerForDoc(element);
 
     Services.videoManagerForDoc(element).register(this);
+
+    // Add cache sources if opted-in and it's sources are not already cached.
+    if (
+      this.element.hasAttribute('cache') &&
+      !this.element.querySelector('source[amp-orig-src]')
```

Comment:
Why this custom check over hasAnyCachedSources?

Reply:
We can use `hasAnyCachedSources`